include/vcpkg/vcpkgcmdarguments.h

```diff
@@ -187,10 +188,26 @@ namespace vcpkg
         constexpr static StringLiteral ASSET_SOURCES_ENV = "X_VCPKG_ASSET_SOURCES";
         constexpr static StringLiteral ASSET_SOURCES_ARG = "asset-sources";
 
+        constexpr static StringLiteral GITHUB_RUN_ID_ENV = "GITHUB_RUN_ID";
+        Optional<std::string> github_run_id;
+        constexpr static StringLiteral GITHUB_TOKEN_ENV = "GITHUB_TOKEN";
+        Optional<std::string> github_token;
+        constexpr static StringLiteral GITHUB_JOB_ENV = "GITHUB_JOB";
+        Optional<std::string> github_job;
+        constexpr static StringLiteral GITHUB_WORKFLOW_ENV = "GITHUB_WORKFLOW";
+        Optional<std::string> github_workflow;
+        constexpr static StringLiteral GITHUB_WORKSPACE_ENV = "GITHUB_WORKSPACE";
+        Optional<std::string> github_workspace;
+        constexpr static StringLiteral DEPENDENCY_GRAPH_VERSION_ENV = "DEPENDENCY_GRAPH_VERSION";
+        Optional<std::string> dependency_graph_version;
```

Comment:
It seems like this shouldn't be an input... after all, the format that we are emitting is fixed, we aren't looking to match any particular version.

Reply:
There might be situations where we'd need to emit different versions of input to the Dependency Graph API.

Dependency graph (and presumably the submission API) is available on GitHub Enterprise Server, which is their self-hosted product, which means that some users might be working against an older version of the API.  I don't have a good grasp on the likelihood of that, but it's a possibility.

Reply:
@michaelbprice OK, we might have to respond to the version. But this isn't doing that; it's always emitting the same schema and just claiming it's version whatever github asked for.

Reply:
I'm not sure it works like that. I believe that the request you send to the GitHub API includes the version of the API that the requestor wants to talk. The GitHub API will respond with a success or failure. I presume if the failure is due to an API version that the endpoint doesn't support, the error returned would indicate that.

@dan-shaw Is having the Dependency graph submission API version be a configurable useful, if we don't also have different JSON content to send if the user specifies a different version? I guess it might be useful to reserve the environment variable, and maybe by going ahead and doing the reading, we might be planning for the future, but it has no real impact (other than exposing another potential attack vector) currently, right?

Reply:
The formatting isn't fixed, we set the version of the snapshot based on `DEPENDENCY_GRAPH_VERSION`

Reply:
>I believe that the request you send to the GitHub API includes the version of the API that the requestor wants to talk.

Right, I'm saying we need to listen to and respond to that, not just blindly claim that we know how to make v.future's format.

Reply:
That is, it looks like we are claiming that we want to talk in whatever version github claims to understand. @dan-shaw pointed me to docs on this: https://docs.github.com/en/rest/dependency-graph/dependency-submission?apiVersion=2022-11-28

The text for `version` is:

> The version of the repository snapshot submission.

which suggests to me the version of the submission, not the version of the code for which the submission is. The example on that page always passes `0`. Do we know anyone at GitHub who can clarify what this means?

Reply:
Pinging @courtneycl and/or @pcarlisle

Reply:
👋🏽 hello from Dependency Graph! We could probably have been more clear in the documentation, so I apologize for that.

This refers to the version of the endpoint/schema, which is fixed at `0` right now. There's no reason to use any other value for the foreseeable future, so feel free to hard-code it.